transformer/transformer.go

```diff
@@ -0,0 +1,289 @@
+	DescriptionEdition   []string `json:"description.edition"`
+	DescriptionSummary   []string `json:"description.summary"`
+	DescriptionMeta      []string `json:"description.metaDescription"`
+	DescriptionKeywords  []string `json:"description.keywords"`
+	DescriptionDatasetID []string `json:"description.datasetId"`
+}
+type esResponseAggregations struct {
+	DocCounts struct {
+		Buckets []esBucket `json:"buckets"`
+	} `json:"docCounts"`
+}
+type esBucket struct {
+	Key   string `json:"key"`
+	Count int    `json:"doc_count"`
+}
+
+// New returns a new instance of Transformer
+func New() *Transformer {
+	return &Transformer{}
+}
+
+// TransformSearchResponse transforms an elastic search response into a structure that matches the v1 api specification
+func (t *Transformer) TransformSearchResponse(ctx context.Context, responseData []byte) ([]byte, error) {
+
+	var esResponse esResponse
+
+	err := json.Unmarshal(responseData, &esResponse)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to decode elastic search response")
+	}
+
+	if len(esResponse.Responses) < 1 {
+		return nil, errors.New("Response to be transformed contained 0 items")
+	}
+
+	sr := searchResponse{
+		Count:        esResponse.Responses[0].Hits.Total,
+		Items:        []contentItem{},
+		ContentTypes: []contentType{},
+	}
+	var took int = 0
+	for _, response := range esResponse.Responses {
+		for _, doc := range response.Hits.Hits {
+			sr.Items = append(sr.Items, buildContentItem(doc))
+		}
+		for _, bucket := range response.Aggregations.DocCounts.Buckets {
+			sr.ContentTypes = append(sr.ContentTypes, buildContentTypes(bucket))
+		}
+		took += response.Took
+	}
+	sr.Took = took
+
+	transformedData, err := json.Marshal(sr)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to encode transformed response")
+	}
+	return transformedData, nil
+}
+
+func buildContentItem(doc esResponseHit) contentItem {
+	ci := contentItem{
+		Description: buildDescription(doc),
+		Type:        doc.Source.Type,
+		URI:         doc.Source.URI,
+		Matches:     buildMatches(doc.Highlight),
+	}
+
+	return ci
+}
+
+func buildDescription(doc esResponseHit) description {
+	sd := doc.Source.Description
+	return description{
+		Summary:           sd.Summary,
+		NextRelease:       sd.NextRelease,
+		Unit:              sd.Unit,
+		Keywords:          sd.Keywords,
+		ReleaseDate:       sd.ReleaseDate,
+		Edition:           sd.Edition,
+		LatestRelease:     sd.LatestRelease,
+		Language:          sd.Language,
+		Contact:           sd.Contact,
+		DatasetID:         sd.DatasetID,
+		Source:            sd.Source,
+		Title:             sd.Title,
+		MetaDescription:   sd.MetaDescription,
+		NationalStatistic: sd.NationalStatistic,
+		Headline1:         sd.Headline1,
+		Headline2:         sd.Headline2,
+		Headline3:         sd.Headline3,
+	}
+}
+
+func buildMatches(hl esHighlight) matches {
+	var matches matches
+
+	for _, m := range hl.DescriptionTitle {
```

Comment:
Once the struct changes to use omitempty, this will need a check if field is nil

This will also apply to other fields in `esHighlight` object

Reply:
Updated to handle empty matches.